dataplane/loader/src/main.rs

```diff
@@ -16,12 +17,51 @@ use clap::Parser;
 use common::{BackendKey, BackendList, ClientKey, LoadBalancerMapping};
 use log::{info, warn};
 
+/// Command-line options for the application.
+///
+/// This struct defines the options available for the command-line interface,
+/// including an interface name (`iface`) and an optional TLS configuration (`tls_config`).
 #[derive(Debug, Parser)]
 struct Opt {
+    /// Name of the network interface to attach the eBPF programs to.
+    ///
+    /// By default, this is set to `"lo"` (the loopback interface).
     #[clap(short, long, default_value = "lo")]
     iface: String,
+    /// Optional TLS configuration for securing the API server.
+    ///
+    /// If no TLS configuration is provided, the server will start without TLS.
+    /// You can specify either `tls` for server-only TLS or `mutual-tls` for mutual TLS.
+    #[clap(subcommand)]
+    tls_config: Option<TLSConfig>,
 }
 
+/// Main function for the application.
+///
+/// This function sets up and runs eBPF programs on the specified network interface
+/// and optionally configures TLS for the API server.
+///
+/// The program supports an optional TLS configuration, allowing the user to choose between:
+/// - `tls`: Server-only TLS.
+/// - `mutual-tls`: Mutual TLS, where both server and client authenticate with certificates.
+///
+/// # Arguments
+///
+/// - `iface`: The network interface to attach the eBPF programs to.
+/// - `tls_config`: Optional subcommand to configure TLS for the API server.
+///
+/// # Example
+///
+/// ```bash
+/// # Running with default interface and no TLS config:
+/// $ dataplane
+///
+/// # Running with a specified interface and server-only TLS config:
+/// $ dataplane --iface eth0 tls --server-certificate-path /path/to/cert --server-private-key-path /path/to/key
+///
+/// # Running with mutual TLS config:
+/// $ dataplane --iface eth0 mutual-tls --server-certificate-path /path/to/cert --server-private-key-path /path/to/key --client-certificate-authority-root-path /path/to/ca
+/// ```
```

Comment:
Your taking the time to add documentation here is greatly appreciated :bow: :+1: